src/components/@shared/Onboarding/index.tsx

```diff
@@ -46,7 +46,8 @@ export default function OnboardingSection(): ReactElement {
   const { balance } = useBalance()
   const web3Provider = useProvider()
   const { chain } = useNetwork()
-  const { onboardingStep, setOnboardingStep } = useUserPreferences()
+  const { onboardingStep, setOnboardingStep, setShowOnboardingModule } =
+    useUserPreferences()
```

Comment:
suggestion(blocking): rather make use of the `useUserPreferences` hook in the `HideButton` component itself to avoid having to pass the `setShowOnboardingModule` function through multiple child components